### ISSUE TITLE
Frameless window not rendering properly after upgrade to High Sierra

### ISSUE THREAD

**Issue by singhshashi** [electron/electron#11105]
* Electron version: 1.7.9
* Operating system: Mac OS High Sierra 10.13.1

### Expected behavior

Frameless window should render properly as a window with frame does. 

![screen shot 2017-11-13 at 1 31 27 pm](https://user-images.githubusercontent.com/371601/32715155-138c49f2-c877-11e7-9d9d-29985aee1673.png)

### Actual behavior

Frameless window appears darkened. White background appears as if there is a shadow on it. 

![screen shot 2017-11-13 at 1 25 27 pm](https://user-images.githubusercontent.com/371601/32715004-650f7bba-c876-11e7-915e-b3ff1594d452.png)

### How to reproduce

On MacOS High Sierra 10.13.1, clone the electron-quick-start repo 

[https://github.com/electron/electron-quick-start](https://github.com/electron/electron-quick-start)

1. Update electron package to latest electron version.

2. Run npm start to see normal behavior.
3. Change window to frameless by adding frame: false in window def. 

Run npm start again and observe frameless window does not render properly.

**Comment by singhshashi** [comment]
I noticed the following message in the main process, which I thought might be related to this issue. 
This did not used to appear earlier.  
`
Electron[79839:10767804] *** WARNING: Textured window <AtomNSWindow: 0x7fb84c5a9540> is getting an implicitly transparent titlebar. This will break when linking against newer SDKs. Use NSWindow's -titlebarAppearsTransparent=YES instead.`

**Comment by steverandy** [comment]
Is there any temporary solution for this?

**Comment by singhshashi** [comment]
I just observed that the rendering issue is observed even when I use titleBarStyle:'hidden', instead of frame:false

**Comment by hpohlmeyer** [comment]
I am not sure, if this is a bug.
The color is undefined and I think it defaults to whatever is defined as the default color in the underlying native code.

You can fix it by explicitly defining the color in the window options:

```js
mainWindow = new BrowserWindow({
  width: 800,
  height: 600,
  frame: false, // or titleBarStyle: 'hidden'
  backgroundColor: '#fff',
})
```

But I imagine, that a white default color could prevent confusion …

**Comment by singhshashi** [comment]
Thanks @hpohlmeyer. 😃  That was helpful and allows me to move ahead. But this is a bug because this behavior was not observed in older versions of Mac. Also, before the background color comes into effect, the gray overlay still shows

http://www.giphy.com/gifs/xULW8pXttvb5ikhOsU

This is something which the Electron should fix.

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v2.0.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons.

Thanks in advance! Your help is appreciated.

**Comment by singhshashi** [comment]
Sure, I will try and repro this on the latest versions of electron and report back in a few days.

**Comment by sofianguy** [comment]
We are no longer releasing bug fixes for versions earlier than 2.0.x. We'll be happy to reopen if this bug is present in versions later than 2.0.x

**Comment by sloantash** [comment]
This bug is still present in the current version of Electron. 3.1.6
